### PR TITLE
Dispose of stream to allow cleanup of open connections

### PR DESCRIPTION
**Pull request by distantcam**
This should help with the open connections left over from a closed server.

### CODE REVIEW

File: source/Halibut/Transport/SecureListener.cs [hunk 1]
```diff
@@ -200,7 +200,17 @@ async Task ExecuteRequest(TcpClient client)
...
                     if (Authorize(ssl, clientName))
                     {
+                        // The ExchangeMessage call can hang on reading the stream which keeps a thread alive,
+                        // so we dispose the stream which will cause the thread to abort with an exceptions.
+                        var weakSSL = new WeakReference(ssl);
+                        cts.Token.Register(() =>
+                        {
+                            if (weakSSL.IsAlive)
+                                ((IDisposable)weakSSL.Target).Dispose();
+                        }, true);
```
Comment (johnsimons) [inline source/Halibut/Transport/SecureListener.cs]: Does this need to be `true`?

File: source/Halibut/Transport/SecureListener.cs [hunk 2]
```diff
@@ -200,7 +200,17 @@ async Task ExecuteRequest(TcpClient client)
...
+
                         // Delegate the open stream to the protocol handler - we no longer own the stream lifetime
+                        //await Task.WhenAny(cts.Token.WhenCanceled(), ExchangeMessages(ssl));
```
Comment (johnsimons) [inline source/Halibut/Transport/SecureListener.cs]: Not needed?

Comment (distantcam) [reply source/Halibut/Transport/SecureListener.cs]: Whoops. Leftover from other things I was trying.

File: source/Halibut/Transport/SecureListener.cs (hunk 1 above)
Comment (distantcam) [reply source/Halibut/Transport/SecureListener.cs]: Nope. Will change.

**Pull request merged**